stan/math/rev/core/var.hpp

```diff
@@ -295,6 +295,13 @@ class var_value<T, require_floating_point_t<T>> {
   }
 };
 
+namespace internal {
+template <typename T>
+using require_matrix_var_value = require_t<bool_constant<
```

Comment:
I'm not a huge fan of defining additional `require_`s within individual headers, but if you don't see this being used anywhere else then it's fine to leave

Reply:
So this one is very particular. I'd like to leave it here till unless we need it in other places in the future. This was just in the `vari<>` template at first and hasn't been used anywhere else. I just pulled it out because it's a bit cleaner looking.